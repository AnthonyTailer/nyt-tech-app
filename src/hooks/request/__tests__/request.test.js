import news, { initialState } from '../reducer'
import { fetching, success, error } from '../actionCreators'


it('returns the same state on an unhandled action', () => {
  expect(news(initialState, {type: '_NULL'})).toMatchSnapshot()
});

it('handles ERROR action', () => {
  const e = new Error('Look ma! I am an error');
  expect(news(initialState, error(e))).toMatchSnapshot()
});

it('handles FETCHING action', () => {
  expect(news(initialState, fetching())).toMatchSnapshot()
});

it('handles SUCCESS action', () => {
  expect(news(initialState, success())).toMatchSnapshot()
});
