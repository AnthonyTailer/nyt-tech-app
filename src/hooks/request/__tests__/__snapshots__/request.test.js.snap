// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles ERROR action 1`] = `
Object {
  "response": [Error: Look ma! I am an error],
  "status": "error",
}
`;

exports[`handles FETCHING action 1`] = `
Object {
  "response": null,
  "status": "fetching",
}
`;

exports[`handles SUCCESS action 1`] = `
Object {
  "response": undefined,
  "status": "success",
}
`;

exports[`returns the same state on an unhandled action 1`] = `
Object {
  "response": null,
  "status": null,
}
`;
